test/e2e_yaml_test.go

```diff
@@ -0,0 +1,238 @@
+// +build e2e,yaml
+
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
```

Comment:
is it possible to put this test into the same dir as the examples? would be great for ppl to be able to discover it easily :D

Reply:
I like the idea! :)

Unfortunately I'm having trouble placing the example tests into the `examples` directory due to dependencies on some of the test helpers in the `test` directory.

I've tried invoking `go test -v -count=1 -tags=examples -timeout=20m ./examples/`, however I end up with undefined warnings:

```
# github.com/tektoncd/pipeline/examples [github.com/tektoncd/pipeline/examples.test]
examples/examples_test.go:57:51: undefined: clients
examples/examples_test.go:58:9: undefined: WaitForPipelineRunState
examples/examples_test.go:58:73: undefined: Succeed
examples/examples_test.go:66:47: undefined: clients
examples/examples_test.go:68:9: undefined: WaitForTaskRunState
examples/examples_test.go:68:45: undefined: Succeed
examples/examples_test.go:108:41: undefined: clients
examples/examples_test.go:116:37: undefined: clients
examples/examples_test.go:124:19: undefined: setup
examples/examples_test.go:145:19: undefined: setupWithOverrides
examples/examples_test.go:145:19: too many errors
FAIL    github.com/tektoncd/pipeline/examples [build failed]
FAIL
```

When I attempt `go test -v -count=1 -tags=examples -timeout=20m .`, I see:

```
can't load package: package .: no Go files in .../src/github.com/tektoncd/pipeline
```

Do you have any suggestions for building the `examples` directory while also including dependencies from `test`?